cray_product_catalog/catalog_delete.py

```diff
@@ -61,19 +61,153 @@
+    def product_name(self, __product_name):
+        self.__product_name = __product_name
+
+    @property
+    def product_version(self):
+        return self.__product_version
+
+    @product_version.setter
+    def product_version(self, __product_version):
+        self.__product_version = __product_version
+
+    @property
+    def max_reties_for_main_cm(self):
+        return self.__max_reties_for_main_cm
+
+    @max_reties_for_main_cm.setter
+    def max_reties_for_main_cm(self, __max_reties_for_main_cm):
+        self.__max_reties_for_main_cm = __max_reties_for_main_cm
+
+    @property
+    def max_reties_for_prod_cm(self):
+        return self.__max_reties_for_prod_cm
+
+    @max_reties_for_prod_cm.setter
+    def max_reties_for_prod_cm(self, __max_reties_for_prod_cm):
+        self.__max_reties_for_prod_cm = __max_reties_for_prod_cm
+
+    @property
+    def key(self):
+        return self.__key
+
+    @key.setter
+    def key(self, __key):
+        self.__key = __key
+
+    @property
+    def main_cm_fields(self):
+        return self.__main_cm_fields
+
+    @main_cm_fields.setter
+    def main_cm_fields(self, __main_cm_fields):
+        self.__main_cm_fields = __main_cm_fields
+
+    @property
+    def product_cm_fields(self):
+        return self.__product_cm_fields
+
+    @product_cm_fields.setter
+    def product_cm_fields(self, __product_cm_fields):
+        self.__product_cm_fields = __product_cm_fields
+
+    # private methods
+    def __key_belongs_to_main_cm_fields(self):
+        return self.__key in self.__main_cm_fields
+
+    def __key_belongs_to_prod_cm_fields(self):
+        return self.__key in self.__product_cm_fields
+
+    def __modify_main_cm(self):
+        LOGGER.info(f"Removing from config_map={self.main_cm} in namespace={self.cm_namespace} "
+                    f"for {self.product_name}/{self.product_version} (key={self.key})")
+        modify_config_map(self.__main_cm, self.__cm_namespace, self.__product_name, self.__product_version,
+                          self.__key, self.__max_reties_for_main_cm,)
+
+    def __modify_product_cm(self):
+        LOGGER.info(f"Removing from config_map={self.product_cm} in namespace={self.cm_namespace} "
+                    f"for {self.product_name}/{self.product_version} (key={self.key})")
+        modify_config_map(self.__product_cm, self.__cm_namespace, self.__product_name, self.__product_version,
+                          self.__key, self.__max_reties_for_prod_cm,)
+
+    # public method
+    def modify_config_map(self):
+        if self.__key:
+            if self.__key_belongs_to_main_cm_fields():
+                self.__modify_main_cm()
+
+            elif self.__key_belongs_to_prod_cm_fields():
+                self.__modify_product_cm()
+
+            else:
+                LOGGER.error(f"Invalid KEY={self.key} is input so exiting...")
+                return
+
+        else:
+            self.__modify_main_cm()
+            self.__modify_product_cm()
```

Comment:
Moving the `return` call to the end of the main -then clause means that we can simplify the code by getting rid of the `else` statement.
```suggestion
            return

        self.__modify_main_cm()
        self.__modify_product_cm()
```

Reply:
Modified as stated